### ISSUE TITLE
the model is not saved to where model_file specify

### ISSUE THREAD

**Issue by Ericzh2017** [facebookresearch/ParlAI#1730]
I have run `~/ParlAI/projects/wizard_of_wikipedia/generator/train_end2end.py` with a parameter "model_file" specified as  'tmp/model'. But the dir `"~/ParlAI/projects/wizard_of_wikipedia/generator/tmp"` is empty after training.
I can reload the model to train again if I run "train_end2end.py" without modification again but I fail to eval it because I can not find out where the checkpoint file is written. 
Where is the checkpoint file saved?

**Comment by stephenroller** [comment]
Did you possibly specify `/tmp/model` rather than `tmp/model`? Most of our training scripts root on `/tmp`.

**Comment by stephenroller** [comment]
Also, I'm not entirely sure whether or not we call `mkdir` if a directory doesn't exist, but I don't think we do.

**Comment by Ericzh2017** [comment]
Thanks for answering.  I have set model_file to '/tmp/model'.  And I have used "mkdir tmp" to create an empty tmp explicitly in command line before training. "/ParlAI/projects/wizard_of_wikipedia/generator/tmp" is the only dir including"tmp" from "/ParlAI".
It is wired that not file is saved to where "model_file" point to.


if __name__ == '__main__':
    parser = setup_args()
    parser.set_defaults(
        task='wizard_of_wikipedia:generator:random_split',
        model='projects.wizard_of_wikipedia.generator.agents:EndToEndAgent',
        model_file='/tmp/model',
        dict_lower=True,
        dict_tokenizer='bpe',
        n_layers=5,
        n_heads=2,
        dropout=0.20,
        ffn_size=512,
        embeddingsize=256,
        log_every_n_secs=10,
        validation_patience=12,
        validation_metric='ppl',
        validation_metric_mode='min',
        validation_every_n_epochs=0.5,
        n_positions=128,
        truncate=128,
        max_knowledge=32,
        knowledge_alpha=0.95,
        knowledge_truncate=32,
        learningrate=5e-4,
        warmup_updates=5000,
        clip=0.1,
        lr_scheduler='invsqrt',
        embedding_type='fasttext',
        beam_size=1,
        skip_generation=False,
        batchsize=32,
    )
    TrainLoop(parser.parse_args()).train()

**Comment by stephenroller** [comment]
Then it will appear in the absolute directory `/tmp`, not the relative directory `tmp`. Please look in /tmp or remove the front slash.

**Comment by Ericzh2017** [comment]
found it. thanks a lot
